hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/ClientManagerImpl.java

```diff
@@ -128,6 +136,13 @@ private CallableRaisingIOE<S3AsyncClient> createAyncClient() {
         () -> clientFactory.createS3AsyncClient(getUri(), clientCreationParameters));
   }
 
+  private CallableRaisingIOE<S3Client> createUnencryptedS3Client() {
```

Comment:
Needs better handling of deployments without this, so raising IllegalStateException when getOrCreateUnencryptedS3Client() is called. With a test.

Reply:
createUnencryptedS3Client will be only called when v1 compatibility is used.